### PR TITLE
Sample and serialization IO optimizations

### PR DESCRIPTION
**Pull request by benjaminpkane**
To be merged after #3152 

### Changes

* Adds index usage in the sidebar filters to sample fields via `fiftyone.server.view_make_query`
  * Adds index usage in the label tags sidebar filter via `fiftyone.server.view._match_label_tags`
  * Does **not** include index usage for Keypoint fields as their filtering is unique. Will add keypoint support in a separate PR
  * Does **not** include index usage for frame fields because the filtering occurs after the lookup stage. But they do utilize the new matching stages produced by `fiftyone.server.view_make_query`

* Adds the exclusion of all `DictField` data in the grid pagination samples `fiftyone.server.view._project_pagination_paths`
  * Frame field data is also excluded, but it occurs after the frame collection lookup 
  * Moves modal sample data to it's own network request so the full sample document is included

Notable cleanup includes removing label copying when filtering in the grid in favor of a separate modal request and to add index usage and new components for modal navigiation.

### CODE REVIEW

**Review by manivoxel51 — COMMENTED** [review]
nice work Ben! 🍨 

left couple of comments

**Comment by benjaminpkane** [comment]
> Found some bugs:
> 
> 1. Tag some labels in the sample modal view, the modal will close and re-opening the sample modal has an error `Variable '$form' got invalid value None at 'form.sampleIds[0]'; Expected non-nullable type 'ID!' not to be None`.
>    https://github.com/voxel51/fiftyone/assets/17770824/a0facba7-f214-4af4-a4e1-5646ee6dd6f2
> 2. Select samples in the modal and get back to grid view, the selected samples are not shown as selected in the grid.
>    https://github.com/voxel51/fiftyone/assets/17770824/78a21e4f-838b-4afd-bbbb-3fcecf1eae4f
> 3. Select some samples and then click the tagger, the build break and jump back to grid view.
>    https://github.com/voxel51/fiftyone/assets/17770824/34cf5e45-aa26-4d77-b3a0-879eccfc9ea3

Yes, I am still finalizing things

File: app/packages/relay/src/queries/mainSample.ts [hunk 1]
```diff
@@ -7,9 +7,10 @@ export default r(graphql`
     $dataset: String!
     $view: BSONArray!
     $filter: SampleFilter!
-    $index: Int!
+    $filters: JSON
```
[the remark this answers is not in the document]
Comment (benjaminpkane) [reply app/packages/relay/src/queries/mainSample.ts:10]: Yes, it's referring to sidebar filters

File: app/packages/state/src/recoil/modal.ts [hunk 2]
```diff
@@ -10,3 +18,81 @@ export const sidebarSampleId = selector({
...
+    return current ? current.id : null;
+  },
+});
+
+export const isModalActive = selector<boolean>({
+  key: "isModalActive",
+  get: ({ get }) => get(currentModalSample) !== null,
+});
+
+export const modalSample = graphQLSelector<
```
[the remark this answers is not in the document]
Comment (benjaminpkane) [reply app/packages/state/src/recoil/modal.ts:75]: We definitely still use `graphQLSelector` in OSS still. I don't see an issue with it if the loading state is meant to independent from other state changes as it is an independent query. But I am already working on moving this to a custom selector that integrates with relay for sample routing. This is just to get optimizations out, as the modal had to be moved to a separate request, and a lot of modal data flow already runs through recoil

**Pull request merged**